public/js/NeighbourImages.js

```diff
@@ -83,48 +102,83 @@ NeighbourImages.prototype.getImages = function(sample_id) {
  */
 NeighbourImages.prototype.setImages = function() {
     var self = this;
-    var $nebula0 = $('#nebula-0');
-    var $body = $('body');
-
-    console.log(self.neighbourImages);
 
     // make all img frames empty
-    $nebula0
+    this.selectors.$nebula0
         .attr('src', '//:0')
         .attr('title', 'not found');
 
-    $('.thumbnail > img')
+    this.selectors.$thumbnailChildImg
         .attr('src', '//:0')
         .attr('title', 'not found');
 
     // add selected main image, if it exists
-    if(self.selectedImage) {
-        $nebula0
+    if(this.selectedImage) {
+        this.selectors.$nebula0
             .attr('src', 'data:image/jpg;base64,' +
-                self.selectedImage.image_large)
-            .attr('title', self.selectedImage.sample_id);
+                this.selectedImage.image_large)
+            .attr('title', this.selectedImage.sample_id);
     }
 
-    $nebula0.on('click', function(event) {
-        // prevent browser from scrolling to top when main image clicked on
+    this.selectors.$imgZoomButton.unbind('click');
+    this.selectors.$imgZoomButton.on('click', function(event) {
         event.preventDefault();
+        self.openFullModal();
     });
 
     // iterate through all found images and add them to the gallery
     for(var i = 0; i < self.neighbourImages.length; i++) {
-        var $nebulaSelector = $('#nebula-' + (i+1));
-        $nebulaSelector
+
+        this.selectors.$nebula[i]
             .attr('src', 'data:image/jpg;base64,' +
                 self.neighbourImages[i].image_thumb)
             .attr('title', self.neighbourImages[i].sample_id);
-        $nebulaSelector.unbind('click');
+        this.selectors.$nebula[i].unbind('click');
+
+        // use IIFE here as closure being called in a loop
+        // see http://www.mennovanslooten.nl/blog/post/62
         (function(j) {
-            $nebulaSelector.on('click', function(event) {
+            self.selectors.$nebula[i].on('click', function(event) {
                 event.preventDefault();
-                $body.trigger('updatePoint', self.neighbourImages[j].sample_id);
+                self.selectors.$body.trigger('updatePoint',
+                    self.neighbourImages[j].sample_id);
             });
         })(i);
     }
 };
 
+/**
+ * openFullModal: Open the modal that displays the full size image.
+ *
+ * Sends an AJAX request to get the full-size image, sets the content of
+ * the image modal and opens it.
+ *
+ * @this {NeighbourImages}
+ */
+NeighbourImages.prototype.openFullModal = function() {
+    var imageQuery = {
+        sample_id: this.selectedImage.sample_id,
+        select: ['image_full image_large sample_id']
+    };
+    $.ajax({
+        type: 'GET',
+        url: '/api/images/?' + $.param(imageQuery),
+        success: function(data) {
+            // TODO some collections don't have a fullsize image, use the large image in lieu
```

Comment:
Hmm. Ok that's fair. Maybe add a line: "remove once all collections reliably contain fullsize"